akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolConductor.scala

```diff
@@ -129,11 +130,20 @@ private object PoolConductor {
       val slotStates = Array.fill[SlotState](slotSettings.maxSlots)(Unconnected)
       var nextSlot = 0
 
+      def updateSlotState(idx: Int, f: SlotState ⇒ SlotState): Unit = {
+        val oldState = slotStates(idx)
+        val newState = f(oldState)
+
+        debug(s"[$idx] $oldState -> $newState")
```

Comment:
Why string interpolation? We advice against it since this will always build the string even if debug level is off.
Use `log.debug("... {}", ...)` instead